plugins/GWSplits/GWSplits.cpp

```diff
@@ -559,13 +665,54 @@ void GWSplits::Draw(IDirect3DDevice9* pDevice)
             ImGui::TableSetupColumn("Relative", ImGuiTableColumnFlags_WidthFixed);
             ImGui::TableSetupColumn("Tracked", ImGuiTableColumnFlags_WidthFixed);
 
-            int row = 0;
+            int currentSplitIndex = currentSplitIt - currentSplits.begin();
+
+            auto displaySplits = std::span(currentSplits);
+
+            if (isCurrentRunTracked && totalSplits > 0 && totalSplits < (int) currentSplits.size()) {
+                int total = (int) currentSplits.size();
+
+                int end = std::min(currentSplitIndex + std::max(upcomingSplits, 0), total - 1);
+                int upcomingCount = end - currentSplitIndex;
+                int start = currentSplitIndex - (totalSplits - 1 - upcomingCount);
+
+                displaySplits = displaySplits.subspan(std::max(start, 0), std::min(totalSplits, total));
+            }
+
+            int previousSplitTime = 0;
 
-            for (const auto& split : currentSplits) {
+            for (const auto& split : displaySplits) {
                 ImGui::TableNextRow();
-                if (row == (currentSplitIt - currentSplits.begin())) 
-                    ImGui::TableSetBgColor(ImGuiTableBgTarget_RowBg1, ImGui::GetColorU32(ImVec4{0.3725f, 0.3961f, 0.7529f, 0.6f}));
+
                 ImGui::TableNextColumn();
+                if (split.completed) {
+                    ImGui::PushID(&split);
+                    ImGui::Selectable("##ping_split###ID", false, ImGuiSelectableFlags_SpanAllColumns | ImGuiSelectableFlags_AllowOverlap);
+
+                    if (ImGui::IsItemHovered()) {
+                        ImGui::SetTooltip("Click to send to team chat");
+                    }
+
+                    if (ImGui::IsItemClicked()) {
+                        std::unordered_map<std::string, std::string> variables = {
+                            {"name", split.name},
+                            {"time", timeToString(split.currentTime)},
+                            {"duration", timeToString(split.currentTime - previousSplitTime)},
+                            {"diff", timeToString(split.currentTime - split.trackedTime, ToStringStyle::SecondsCentiseconds)},
+                            {"reference", timeToString(split.trackedTime)},
+                            {"pb", timeToString(split.pbSegmentTime)},
+                        };
+                        auto result = replacePlaceholders(splitMessage, variables);
```

Comment:
```suggestion
                        auto result = replacePlaceholders(splitMessage, std::move(variables));
```
FYI you could avoid copying the map here... not that it really matters in this case given that it's small